### ISSUE TITLE
Can this be used as a warbler drop in replacement?

### ISSUE THREAD

**Issue by dolzenko** [rammpeter/jarbler#1]
Hi Peter! I'm wondering if this gem can do the similar job to what warbler does with, e.g. `warble runnable compiled war`. This might certainly be useful for people with newer Bundler (2.x) for which warbler seems to be broken recently

**Comment by rammpeter** [comment]
Hi, I started using this tool as a simple replacement for warbler to pack Rails applications in a self executing jar.
Compiling the Ruby sources with jRuby hasn't been on the agenda for the first step, but isn't impossible.
I'll let this issue open as as reminder to add compiling later. (or someone contributes)

**Comment by rammpeter** [comment]
Compilation of .rb files to Java class files is added in rel. 0.2.2 now.
Set "compile_ruby_files=true" in config/jarble.rb to force compilation.
Control compilation of gems by "include_gems_to_compile=true|false"
Exclude certain dirs or files from compilation by "excludes_from_compile"

**Comment by simplytech** [comment]
Hi Peter!
I'm interested to understand the difference between Jarbler and Warbler. Reading this issue, it is still not clear to me when would I choose one tool versus the other. Could you please clarify?

**Comment by rammpeter** [comment]
> Hi Peter! I'm interested to understand the difference between Jarbler and Warbler. Reading this issue, it is still not clear to me when would I choose one tool versus the other. Could you please clarify?

Hi Alain,
there is a huge overlap in function between Warbler and Jarbler, but Jarbler only provides a subset of Warbler's functions.

I used Warbler for several years to create war files with included Jetty app server. 
But with current releases of JRuby it became more and more difficult to keep Warbler functional.
At the end there was less hope that Warbler would be continued and problems would be fixed.

Fixing Warbler myself seamed a bit too complex for me and so I created Jarbler with a limited function:
- Create a self-executable jar file containing a JRuby application with all it's dependencies including the JRuby runtime
- Direct execution of JRuby code without any additional Java JEE or JSP etc. application server runtime (because e.g. Puma covers my requirements completely)
- Execution of JRuby code in the same single JVM instance that is started by "java -jar xyz.jar" to ensure full control of the JVM by Java and JRuby property settings 
- Possibility to precompile Ruby files and this way hide your proprietary source code

So if Warbler works for you, then there might not be a need to switch to Jarbler.
Otherwise, Jarbler is a bit more focused to one purpose.
